File: day1/2/main.py
sum = 0


def isDigit(text):
    if text[:3] == 'one':
        return 1
    if text[:3] == 'two':
        return 2
    if text[:5] == 'three':
        return 3
    if text[:4] == 'four':
        return 4
    if text[:4] == 'five':
        return 5
    if text[:3] == 'six':
        return 6
    if text[:5] == 'seven':
        return 7
    if text[:5] == 'eight':
        return 8
    if text[:4] == 'nine':
        return 9
    return None


for _ in range(1000):
    ln = input()
    i = 0
    j = len(ln) - 1
    c1, c2 = None, None
    while c1 is None:
        try:
            c1 = isDigit(ln[i:])
            c1 = int(ln[i])
        except:
            i += 1

    while c2 is None:
        try:
            c2 = isDigit(ln[j:])
            c2 = int(ln[j])
        except:
            j -= 1
    sum += c1 * 10 + c2
print(sum)

File: day1/2/Main.kt

fun toInt(str: String): Int? {
  try {
    println(str)
    return when {
      str[0].isDigit() -> str[0].digitToInt()
      str.slice(0..3) == "one" -> 1
      str.slice(0..3) == "two" -> 2
      str.slice(0..3) == "six" -> 6
      str.slice(0..4) == "four" -> 4
      str.slice(0..4) == "five" -> 5
      str.slice(0..4) == "nine" -> 9
      str.slice(0..5) == "three" -> 3
      str.slice(0..5) == "seven" -> 7
      str.slice(0..5) == "eight" -> 8
      else -> null
    }
  } catch (e: StringIndexOutOfBoundsException) {
    return null
  }
}

fun main() {
  var sum = 0
  for (x in 1..1000) {
    val ln = readln()
    var i = 0
    var j = ln.length - 1
    var c1: Int? = null
    var c2: Int? = null

    while (c1 == null) {
      c1 = toInt(ln.slice(i..<ln.length))
      i++
    }
    while (c2 == null) {
      c2 = toInt(ln.slice(j..<ln.length))
      j--
    }
    sum += c1 * 10 + c2
  }
  println(sum)
}
